hbase-hbtop/src/main/java/org/apache/hadoop/hbase/hbtop/mode/ModeStrategy.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.hbtop.mode;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.List;
+import org.apache.hadoop.hbase.ClusterMetrics;
+import org.apache.hadoop.hbase.hbtop.Record;
+import org.apache.hadoop.hbase.hbtop.field.Field;
+import org.apache.hadoop.hbase.hbtop.field.FieldInfo;
+import org.apache.yetus.audience.InterfaceAudience;
+
+
+/**
+ * An interface for strategy logic for {@link Mode}.
+ */
+@InterfaceAudience.Private
+interface ModeStrategy {
+  List<FieldInfo> getFieldInfos();
+  Field getDefaultSortField();
+  List<Record> getRecords(ClusterMetrics clusterMetrics);
+  @Nullable DrillDownInfo drillDown(Record selectedRecord);
+}
```

Comment:
How did you come up w/ this model? It looks nice. Makes me wonder though if we couldn't use a lib given much seems generic. But looks great.

Reply:
Not sure there is a lib we can use here. Do you have any pointers?

Reply:
I don't. Just seemed so generic (or to put it another way, nicely modeled), you'd think there'd be libs that did this.

Reply:
Tried looking.

This has right license but doesn't help https://github.com/beryx/text-io Article intro'ing it was kinda good though https://dzone.com/articles/interactive-console-applications-in-java 

Laterna looks ideal but has the licensing problem you already turned up.

http://www.pitman.co.za/projects/charva/Licence.html looked good but similar licensing issue.

https://sourceforge.net/projects/javacurses/files/ is too old.

https://github.com/jline/jline3 has some help but not enough?

Yeah, I don't know enough about this console domain.